### DISCUSSION TITLE
Prosess Framework long lasting process required features

### DISCUSSION THREAD

**Discussion by guru98** [microsoft/semantic-kernel#9105] [General]
I'm looking with interest to the process framework and I think it could be a great idea. In your decision pattern you dedided to build it from scratch instead to leverage existing framework e.g Durable Function. The idea to have a framework specialized to llm integration is great but still I think that it doesn't have some key features: first of all the possibility de-hidradate/re-hidradate the workflow, this is helpfull if you using it in a chat where, for example you re-open the chat or you have a stateless engine like bot framework. Second the possibility to handle long lasting processes where you shold wakeup a workflow when an event occur. In bot cases you should return to the process  suspension point repristinating the status. In my mind there are 2 great examples for those scenarious with different approach: Bot framework with the concept of dialogs and a dialog stack and durable functions. In the documentation you reference use of dapr or orleans to potentially manage those aspects but you should provide examples or go more deeply .

**Comment by alliscode** [comment]
Hi @guru98, thank you for the feedback. The Processes framework will support long-running processes as well as re-hydration of processes. We are working on support for the features that you're asking for now and will be releasing them in the near future. Support for Dapr for example will be out in the next couple of weeks. Stay tuned!

**Reply by waterydan** [reply]
Hi, we are also interested in using process framework for a long running workflow which may take a few minutes or even hours to complete. Is there a sample or recommendation on the best practice? Specifically we're interested in the following:

- The ability to report progress, eg the name of the current running step
- Retry a failed step. Eg if step 5 failed, we'd like to have the ability to retry that step with the output from Step 4. 
- Is there a way to stop a process?

Thanks
